### PR TITLE
[docs] add some classes/methods, fix ref, fix `no_sync` equations

### PR DESCRIPTION
**Pull request by crcrpar**
cc @borda @apaz-cli

### CODE REVIEW

File: thunder/common.py [hunk 1]
```diff
@@ -137,6 +172,17 @@ def last_computation_execution_time(self, /) -> int:
 #   like additional_param_names
 # TODO RC1 Rename this to CompileOptions
 class CompileData:
+    """A class holding data about the compiled object.
+
+    Data include statistics about how it's been called such as ``additional_param_names``.
+
+    Attributes:
+        additional_param_names:
```
Comment (t-vi) [inline thunder/common.py:180]: Let's not have these, they're on their way out.

File: thunder/common.py [hunk 2]
```diff
@@ -53,6 +54,40 @@
 # TODO RC1 Update last_executed to last_computation
 # TODO RC1 Review how autograd traces are presented
 class CompileStats:
+    """A class holding statistics and caches for a compiled function.
+
+    Attributes:
+        last_executed:
```
Comment (t-vi) [inline thunder/common.py:60]: Thank you for looking at these. Could you add a comment that it is preferable to use the top-level thunder.last_traces etc. functions where applicable, please?

File: thunder/common.py (hunk 1 above)
Comment (crcrpar) [reply thunder/common.py:180]: um, like when? really shortly?

Comment (t-vi) [reply thunder/common.py]: Sorry, I missed this. They're only used with thunder.compile.

**Review by t-vi — APPROVED** [review]
Thank you @crcrpar

**Pull request merged**